lib/auth/userloginstate/generator.go

```diff
@@ -154,6 +164,13 @@ func (g *Generator) Generate(ctx context.Context, user types.User) (*userloginst
 		return nil, trace.Wrap(err)
 	}
 
+	// Preserve states like GitHub identities across logins.
+	// TODO(greedy52) implement a way to remove the identity or find a way to
+	// avoid keeping the identity forever.
```

Comment:
Should we set some kind of TTL corresponding to the TTL returned by Github for the auth request?

Reply:
I don't think Github provide any meaningful TTL for the identity itself. we might need to introduce something like `role.max_external_identity_ttl = 168h`

Reply:
Yes, I think we should put a limit here. A week feels like a reasonable compromise.